src/protocol/clock/timeextent.rs

```diff
@@ -0,0 +1,185 @@
+{
+    fn now(increment: &TimeExtentBase) -> Option<Result<TimeExtent, TryFromIntError>> {
+        Clock::now()
+            .as_nanos()
+            .checked_div((*increment).as_nanos())
+            .map(|amount| match TimeExtentMultiplier::try_from(amount) {
+                Err(error) => Err(error),
+                Ok(amount) => Ok(TimeExtent::new(increment, &amount)),
+            })
+    }
+
+    fn now_add(increment: &TimeExtentBase, add_time: &Duration) -> Option<Result<TimeExtent, TryFromIntError>> {
+        match Clock::add(add_time) {
+            None => None,
+            Some(time) => {
+                time.as_nanos()
+                    .checked_div(increment.as_nanos())
+                    .map(|amount| match TimeExtentMultiplier::try_from(amount) {
+                        Err(error) => Err(error),
+                        Ok(amount) => Ok(TimeExtent::new(increment, &amount)),
+                    })
+            }
+        }
+    }
+    fn now_sub(increment: &TimeExtentBase, sub_time: &Duration) -> Option<Result<TimeExtent, TryFromIntError>> {
+        match Clock::sub(sub_time) {
+            None => None,
+            Some(time) => {
+                time.as_nanos()
+                    .checked_div(increment.as_nanos())
+                    .map(|amount| match TimeExtentMultiplier::try_from(amount) {
+                        Err(error) => Err(error),
+                        Ok(amount) => Ok(TimeExtent::new(increment, &amount)),
+                    })
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct TimeExtentMaker<const CLOCK_TYPE: usize> {}
+
+pub type WorkingClockTimeExtentMaker = TimeExtentMaker<{ ClockType::WorkingClock as usize }>;
+pub type StoppedClockTimeExtentMaker = TimeExtentMaker<{ ClockType::StoppedClock as usize }>;
+
+impl MakeTimeExtent<WorkingClock> for WorkingClockTimeExtentMaker {}
+impl MakeTimeExtent<StoppedClock> for StoppedClockTimeExtentMaker {}
+
+#[cfg(not(test))]
+pub type DefaultClockTimeExtentMaker = WorkingClockTimeExtentMaker;
+
+#[cfg(test)]
+pub type DefaultClockTimeExtentMaker = StoppedClockTimeExtentMaker;
+
+#[cfg(test)]
+mod test {
+
+    use std::time::Duration;
+
+    use crate::protocol::clock::timeextent::{DefaultClockTimeExtentMaker, Extent, MakeTimeExtent, TimeExtent};
+    use crate::protocol::clock::{DefaultClock, DurationSinceUnixEpoch, StoppedTime};
+
+    #[test]
+    fn it_should_get_the_total_duration() {
+        assert_eq!(TimeExtent::default().total().unwrap().unwrap(), Duration::ZERO);
+
+        assert_eq!(
+            TimeExtent::from_sec(12, &12).total().unwrap().unwrap(),
+            Duration::from_secs(144)
+        );
+        assert_eq!(
+            TimeExtent::from_sec(12, &12).total_next().unwrap().unwrap(),
+            Duration::from_secs(156)
+        );
+    }
+
+    #[test]
+    fn it_should_make_the_current_extent() {
+        assert_eq!(
+            DefaultClockTimeExtentMaker::now(&Duration::from_secs(2)).unwrap().unwrap(),
```

Comment:
@da2ce7 I think this would be clearer with something like:

```rust
DefaultClockTimeExtentMaker::now(&TimeExtentBase::from_secs(2)).unwrap().unwrap()
```

Reply:
Fixed. :)